providers/resend/src/lib/resend.provider.spec.ts

```diff
@@ -39,3 +39,29 @@ test('should trigger resend library correctly', async () => {
     attachments: mockNovuMessage.attachments,
   });
 });
+
+test('should trigger resend email with From Name', async () => {
+  const mockConfigWithSenderName = {
+    ...mockConfig,
+    senderName: 'Test User',
+  };
+
+  const provider = new ResendEmailProvider(mockConfigWithSenderName);
+  const spy = jest
+    .spyOn(provider, 'sendMessage')
```

Comment:
The tests are currently failing (https://github.com/novuhq/novu/actions/runs/6394994960/job/17358214122?pr=4237#step:5:86) because the `sendMessage` method on the provider is being mocked, instead of the client's send method.

Instead, we should mock the `resendClient`'s `sendEmail` method to execute the tested code. For an example, take a look at the Vonage client tests: https://github.com/novuhq/novu/blob/2f47f7b0add731870bef1672cc928246a654279b/providers/nexmo/src/lib/nexmo.provider.spec.ts#L13

Let me know if you have any trouble with the example and mocking 😀

Reply:
Hi @rifont, Thanks a lot for the example.
I have checkout the implementation of SendGrid and mocked it that way 😄. Tests are now passing on my local machine.

Thanks a lot 🙌🏻